crates/notary/server/src/settings.rs

```diff
@@ -0,0 +1,39 @@
+use config::{Config, ConfigError, Environment, File};
+use std::path::Path;
+use crate::{ CliFields, NotaryServerProperties };
+use serde::Deserialize;
+
+#[derive(Debug, Deserialize)]
+pub struct Settings {
+    #[serde(flatten)]
+    pub config: NotaryServerProperties,
+}
+
+impl Settings {
+    pub fn new(cli_fields: &CliFields) -> Result<Self, ConfigError> {
+        let config_path = Path::new(&cli_fields.config_file);
+
+        let mut builder = Config::builder()
+            // Load base configuration
+            .add_source(File::from(config_path))
+            // Add in settings from environment variables (with a prefix of NOTARY_SERVER and '__' as separator)
+            .add_source(Environment::with_prefix("NOTARY_SERVER").try_parsing(true).prefix_separator("__").separator("_"));

```

Comment:
i just realised we have to use "__" for separator as well; so that things like notarization.max-sent-data can work.

Reply:
Also on this, I am sorry but I understand now why you changed the config.yaml to use snake_case before this, cause else env var won't work with those kebab-case config field..

I was hesitant to change the casing back then since it means we have to change that in our server as well, but I guess that's inevitable now. Can you make that change again 🙏

Reply:
no worries, let me make the necessary changes.